File: algae-common/src/main/java/edu/odu/cs/AlgAE/Common/Communications/AckMessage.java
/**
 * 
 */
package edu.odu.cs.AlgAE.Common.Communications;

/**
 * A generic acknowledgment message, used as a keep-alive in remote servers.
 * 
 * @author zeil
 *
 */
public class AckMessage extends ClientMessage {

	/**
	 */
	public AckMessage() {
		super("Ack");
	}

	@Override
	public boolean equals(Object clientMessage) {
		if (clientMessage == null)
			return false;
		try {
			AckMessage msg = (AckMessage)clientMessage;
			return msg != null;
		} catch (Exception e) {
			return false;
		}
	}

}

File: algae-cppserver/src/main/cpp/ackMessage.cpp
/*
 * ackMessage.cpp
 *
 *  Created on: Jun 10, 2012
 *      Author: zeil
 */


#include <algae/communications/ackMessage.h>
#include <algae/communications/xmlOutput.h>
#include <string>
#include <iostream>
#include <typeinfo>

using namespace std;
using namespace algae;

/**
 * Generic acknowledgment message from server to client.
 *
 *
 * @author zeil
 *
 */

AckMessage::AckMessage()
 : ClientMessage("Ack")
{

}

void AckMessage::printXML (std::ostream& out) const
{
	JavaTag jt (out);
	ObjectTag ot (out, communicationsPackage + ".AckMessage");
	ot.close();
	jt.close();
}

ClientMessage* AckMessage::clone() const
{
	return new AckMessage(*this);
}


bool AckMessage::operator== (const ClientMessage& other) const
{
	if (typeid(other) != typeid(AckMessage))
		return false;
	else
		return true;
}
